app/src/main/kotlin/com/wire/android/ui/calling/common/CallerDetails.kt

```diff
@@ -119,12 +123,44 @@ fun CallerDetails(
                 proteusVerificationStatus
             )
         }
-        Text(
-            text = callingLabel,
-            color = colorsScheme().onBackground,
-            style = MaterialTheme.wireTypography.body01,
-            modifier = Modifier.padding(top = dimensions().spacing8x)
-        )
+
+        FlowRow(
+            modifier = Modifier.padding(
+                top = dimensions().spacing8x,
+                start = dimensions().spacing24x,
+                end = dimensions().spacing24x
+            ),
+            horizontalArrangement = Arrangement.Center,
+        ) {
+            groupCallerName?.let { name ->
+                Text(
+                    text = name,
+                    color = colorsScheme().onBackground,
+                    style = MaterialTheme.wireTypography.body01,
+                    textAlign = TextAlign.Center,
+                    maxLines = 1,
+                    overflow = TextOverflow.Ellipsis,
+                )
+            }
+            val isCallingLabel =
+                if (conversationTypeForCall == ConversationTypeForCall.Conference) {
+                    stringResource(R.string.calling_label_incoming_call_someone_calling)
+                } else {
+                    stringResource(R.string.calling_label_incoming_call)
+                }
+            Text(
+                modifier = Modifier.padding(
+                    start = dimensions().spacing2x,
+                ),
+                text = isCallingLabel,
+                color = colorsScheme().onBackground,
+                style = MaterialTheme.wireTypography.body01,
+                textAlign = TextAlign.Center,
+                maxLines = 1,
+                overflow = TextOverflow.Ellipsis,
+            )
+        }
```

Comment:
Question: not all languages have the verb after the subject, so is there a way to achieve the same and keep it up to the translation to format the sentence? Maybe adding the correct padding and alignment to the original text composable and call it a day?

Reply:
It's not just about padding and alignment, we need to trim and add `ellipsis` for the long name and put "calling" label in a new line.
Curious about those languages having a subject after the verb 😋

Reply:
From compose 1.8.0-alpha2, we will have middle ellipsis (or different types), we can revisit then.
https://issuetracker.google.com/issues/185418980